### ISSUE TITLE
Async media multiupload: cancelling one upload provokes broken pipe for all

### ISSUE THREAD

**Issue by mzorz** [wordpress-mobile/WordPress-Android#6648]
[Comes from this comment](https://github.com/wordpress-mobile/WordPress-Android/pull/6613#issuecomment-331210205)

### Expected behavior
When uploading 2 images on a Post, and canceling one of the uploads what the other continues uploading, the remaining image should continue to upload until it's finished uploading (i.e. canceling one image shouldn't affect any other uploads).

### Actual behavior && Steps to reproduce the behavior
Using a S3 geny image and Charles, setting a breakpoint on the `https://public-api.wordpress.com/rest/v1.1/sites/<siteID>/media/new/` endpoint, and then executing the following steps:

1. start a draft
2. tap on the add media icon
3. select 2 media items to be included, by using the long-press functionality in the media picker
4. the images start uploading, and you'll get the 2 breakpoints (one corresponding to each `/new` call) from Charles, don't do anything as of yet
5. tap on one of the images (I do on the second image always), and when the "Stop uploading?" dialog appears, tap OK.
6. on Charles `Cancel/Abort/Execute` breakpoint prompt, just click `Execute` for all 4 prompts (as it will break execution twice for each breakpoint on connection start and connection end), to simulate no problems with the connection (we just want to test one user-cancelled image).
7. you'll see that from the user's POV the cancelled image goes away (i.e. is deleted) as expected, but shortly after that, the first image shows the "failed" overlay with the retry button.

Looking in Charles, I can see both requests show as failed due to "Broken pipe", 

<img width="835" alt="screen shot 2017-09-21 at 1 19 50 pm" src="https://user-images.githubusercontent.com/6597771/30706820-f92eabc0-9ecf-11e7-8bad-36198527c685.png">

and looking in the logs, you can see this:

````
09-08 16:04:33.915 14559-14559/org.wordpress.android I/WordPress-MEDIA: MediaUploadHandler > 12 - progress: 0.73584414
09-08 16:04:33.915 14559-15006/org.wordpress.android D/WordPress-API: Dispatching action: MediaAction-UPLOADED_MEDIA
09-08 16:04:33.931 14559-14559/org.wordpress.android I/WordPress-MEDIA: MediaUploadHandler > 13 - progress: 0.57438886
09-08 16:04:33.935 14559-14559/org.wordpress.android I/WordPress-MEDIA: MediaUploadHandler > 13 - progress: 0.48602134
09-08 16:04:33.935 14559-14559/org.wordpress.android I/WordPress-MEDIA: MediaUploadHandler > 12 - progress: 0.773999
09-08 16:04:33.979 14559-14559/org.wordpress.android I/WordPress-MEDIA: MediaUploadHandler > 13 - progress: 0.7069401
09-08 16:04:33.983 14559-14559/org.wordpress.android I/WordPress-MEDIA: MediaUploadHandler > 12 - progress: 0.7794497
09-08 16:04:34.083 14559-14559/org.wordpress.android I/WordPress-MEDIA: MediaUploadHandler > 12 - progress: 0.8067032
09-08 16:04:34.083 14559-14559/org.wordpress.android I/WordPress-MEDIA: MediaUploadHandler > 12 - progress: 0.7849004
09-08 16:04:34.083 14559-14559/org.wordpress.android I/WordPress-MEDIA: MediaUploadHandler > 12 - progress: 0.99
09-08 16:04:34.087 14559-14559/org.wordpress.android I/WordPress-MEDIA: MediaUploadHandler > 12 - progress: 0.8394074
09-08 16:04:38.715 14559-14559/org.wordpress.android W/EGL_genymotion: eglSurfaceAttrib not implemented
09-08 16:04:40.347 14559-14559/org.wordpress.android D/WordPress-API: Dispatching action: MediaAction-CANCEL_MEDIA_UPLOAD
09-08 16:04:40.363 14559-15006/org.wordpress.android D/WordPress-MEDIA: Canceled in-progress upload: 2006-06-13_2000x3000_chicago_river.jpg
09-08 16:04:40.363 14559-15006/org.wordpress.android D/WordPress-MEDIA: mediaRestClient: removed id: 13 from current uploads, remaining: 1
09-08 16:04:40.363 14559-15006/org.wordpress.android D/WordPress-API: Dispatching action: MediaAction-CANCELED_MEDIA_UPLOAD
09-08 16:04:40.363 14559-15001/org.wordpress.android W/WordPress-MEDIA: media upload failed: java.io.IOException: Canceled
09-08 16:04:40.387 14559-14559/org.wordpress.android I/WordPress-MEDIA: MediaUploadHandler > Upload successfully canceled
09-08 16:04:40.391 14559-14559/org.wordpress.android W/DebugConnManager: getNetworkInfo() on networkType 1
09-08 16:04:40.399 14559-14559/org.wordpress.android W/DebugConnManager: getNetworkInfo() on networkType 1
09-08 16:04:40.403 14559-14559/org.wordpress.android W/WordPress-MEDIA: MediaUploadHandler > No more media items to upload. Skipping this request.
09-08 16:04:40.407 14559-14559/org.wordpress.android I/WordPress-MAIN: UploadService > Upload cancelled for post with id 22 - a media upload for this post has been cancelled, id: 13
09-08 16:04:44.739 14559-15190/org.wordpress.android D/WordPress-API: Dispatching action: PostAction-UPDATE_POST
09-08 16:04:47.291 14559-14995/org.wordpress.android E/WordPress-MEDIA: error uploading media: 
09-08 16:04:47.291 14559-14995/org.wordpress.android D/WordPress-MEDIA: mediaRestClient: removed id: 12 from current uploads, remaining: 0
09-08 16:04:47.291 14559-14995/org.wordpress.android D/WordPress-API: Dispatching action: UploadAction-UPLOADED_MEDIA
09-08 16:04:47.295 14559-15011/org.wordpress.android D/WordPress-API: Dispatching action: MediaAction-UPLOADED_MEDIA
09-08 16:04:47.303 14559-14559/org.wordpress.android W/WordPress-MEDIA: MediaUploadHandler > Error uploading media: null
09-08 16:04:47.303 14559-14559/org.wordpress.android D/WordPress-API: Dispatching action: MediaAction-UPDATE_MEDIA
09-08 16:04:47.307 14559-14559/org.wordpress.android W/DebugConnManager: getNetworkInfo() on networkType 1
09-08 16:04:47.327 14559-14559/org.wordpress.android W/DebugConnManager: getNetworkInfo() on networkType 1
09-08 16:04:47.335 14559-14559/org.wordpress.android W/WordPress-MEDIA: MediaUploadHandler > No more media items to upload. Skipping this request.
09-08 16:04:47.339 14559-14559/org.wordpress.android I/WordPress-MEDIA: MediaUploadHandler > Completed
09-08 16:04:47.343 14559-14559/org.wordpress.android W/WordPress-MAIN: UploadService > Media upload failed for post 22 : GENERIC_ERROR: null
09-08 16:04:47.347 14559-14559/org.wordpress.android I/WordPress-MAIN: UploadService > Completed
09-08 16:04:47.355 14559-14559/org.wordpress.android W/DebugConnManager: getNetworkInfo() on networkType 1
09-08 16:04:47.371 14559-14559/org.wordpress.android I/WordPress-MAIN: UploadService > Destroyed
09-08 16:04:47.375 14559-14625/org.wordpress.android W/DebugConnManager: getNetworkInfo() on networkType 1
09-08 16:05:44.743 14559-16083/org.wordpress.android D/WordPress-API: Dispatching action: PostAction-UPDATE_POST

````

So, basically, the image shows the failed/retry overlay because it actually failed uploading according to Charles reason being `Broken pipe`. Maybe cancelling one request makes the other request fail as well, for some reason?

**Comment by mzorz** [comment]
Been testing this a while on a couple of real handsets (Nexus 5 and Google Pixel) and been unsuccessful trying to reproduce. It does reproduce for me consistently on the edge case of trying to cancel right before the last part of the image being sent though, and only with the Charles + emulator combination.

Given this, I'm going to close this for now
